# HG changeset patch
# User Rik <rik@octave.org>
# Date 1443553211 25200
#      Tue Sep 29 12:00:11 2015 -0700
# Node ID 2d9ec16fa96026bfb500651ffda2b87833b33ba0
# Parent  ecca283644456cf6ec28b05e8af5530b5aafa9de
Print error, rather than aborting, if mex function mxIsFromGlobalWS is used (bug #46070).

* mex.cc (mxIsFromGlobalWS): Call mexErrMsgTxt rather than abort() in function.

diff --git a/libinterp/corefcn/mex.cc b/libinterp/corefcn/mex.cc
--- a/libinterp/corefcn/mex.cc
+++ b/libinterp/corefcn/mex.cc
@@ -2713,22 +2713,23 @@ mxIsLogicalScalarTrue (const mxArray *pt
 
 // Size predicate.
 int
 mxIsEmpty (const mxArray *ptr)
 {
   return ptr->is_empty ();
 }
 
-// Just plain odd thing to ask of a value.
+// FIXME: Just plain odd thing to ask of a value.
+// Still, Octave is incompatible because it does not implement this.
 int
-mxIsFromGlobalWS (const mxArray */*ptr*/)
+mxIsFromGlobalWS (const mxArray * /*ptr*/)
 {
-  // FIXME
-  abort ();
+  mexErrMsgTxt ("mxIsFromGlobalWS() is unimplemented");
+
   return 0;
 }
 
 // Dimension extractors.
 size_t
 mxGetM (const mxArray *ptr)
 {
   return ptr->get_m ();
